static/src/javascripts/projects/commercial/modules/hosted/onward.js

```diff
@@ -0,0 +1,54 @@
+// @flow
+import config from 'lib/config';
+import mediator from 'lib/mediator';
+import fetchJson from 'lib/fetch-json';
+import fastdom from 'lib/fastdom-promise';
+import HostedCarousel from 'commercial/modules/hosted/onward-journey-carousel';
+
+const loadOnwardComponent = (
+    start: () => void,
+    stop: () => void
+): Promise<any> => {
+    start();
+
+    const placeholders = document.getElementsByClassName(
+        'js-onward-placeholder'
+    );
+
+    if (placeholders.length) {
+        fetchJson(
+            `${config.page.ajaxUrl}/${config.page.pageId}/${config.page.contentType.toLowerCase()}/` +
+                `onward.json`,
+            {
+                mode: 'cors',
+            }
+        )
+            .then(json =>
+                fastdom.write(() => {
+                    let i;
+                    for (i = 0; i < placeholders.length; i += 1) {
```

Comment:
I'm working on a refactor of this module... because this is a bit of a gotcha!

The problem is that although it looks like an array, and mostly acts like an array... placeholders is a [HTMLCollection](https://developer.mozilla.org/en/docs/Web/API/HTMLCollection) full of HTMLElements

Unlike arrays, HTMLCollections do not have a forEach method